src/library/scala/collection/mutable/ArrayBuffer.scala

```diff
@@ -160,6 +160,8 @@ class ArrayBuffer[A] private (initialElements: Array[AnyRef], initialSize: Int)
   def insertAll(@deprecatedName("n", "2.13.0") index: Int, elems: IterableOnce[A]): Unit = {
     checkWithinBounds(index, index)
     elems match {
+      case elems: AnyRef if elems eq this =>
+        insertAll(index, ArrayBuffer.from(this))
```

Comment:
(I just saw https://github.com/scala/bug/issues/12284 / https://github.com/scala/scala/pull/9388. 2.13.5-bin-55f4b15-SNAPSHOT from #9388 shows the same behavior as above).

Reply:
I've been looking at the current implementation of `ArrayBuffer#insertAll` for several minutes now, and I'm struggling to understand why the first example works without this PR. I will continue to investigate, and perhaps remove this case when I can convince myself why it's not needed (and add a note explaining)

Reply:
I believe the reason `b.appendAll(1, b)` works currently is because after copying the end of the array to the right to make space, it doesn't clear the original values from where they were, and `System.arraycopy` "atomically" copies the array into itself.

however, I've discovered that `b.appendAll(1, b.iterator)` (which is the same as `b.appendAll(1, b.view.iterator)`) works currently, even though `b.appendAll(1, b.view)` doesn't, and I have ***no*** idea why.

Reply:
I will the extra case